### PR TITLE
Replace hardcoded -1 by INVALID_ID in AI code

### PR DESCRIPTION
**Pull request by Morlic-fo**

### CODE REVIEW

**Comment by LGM-Doyle** [comment]
Looks all correct to me.

File: default/python/AI/ProductionAI.py [hunk 1]
```diff
@@ -874,15 +875,15 @@ def generate_production_orders():
...
+                    if sys_id == INVALID_ID:
                         continue
                     try:
                         distance_map[sys_id] = universe.jumpDistance(homeworld.systemID, sys_id)
                     except Exception as e:
                         print_error(e, location="ProductionAI.generateProductionOrders")
-                print ([-1] + sorted([(dist, sys_id) for sys_id, dist in distance_map.items()]))
-                use_sys = ([(-1, -1)] + sorted([(dist, sys_id) for sys_id, dist in distance_map.items()]))[:2][-1][-1]  # kinda messy, but ensures a value
-            if use_sys != -1:
+                print ([INVALID_ID] + sorted([(dist, sys_id) for sys_id, dist in distance_map.items()]))
```
Comment (Cjkjvfnby) [inline default/python/AI/ProductionAI.py:884]: Print it as a string `print (["INVALID_ID"]....`

File: default/python/AI/AIDependencies.py [hunk 2]
```diff
@@ -62,6 +62,9 @@
     fo.planetSize.gasGiant: -1,
 }
 
+# C++ code details
```
Comment (Cjkjvfnby) [inline default/python/AI/AIDependencies.py:65]: `This constant is defined in c++ code`

**Review by Cjkjvfnby — APPROVED** [review]
Generally looks ok.

IMHO using `None` or `0` will be more intuitive for python.

**Pull request merged**